packages/relay/src/lib/services/metricService/metricService.ts

```diff
@@ -0,0 +1,297 @@
+   * @param {MirrorNodeClient} mirrorNodeClient - Client for querying the Hedera mirror node.
+   * @param {HbarLimit} hbarLimiter - Rate limiter for managing HBAR-related operations.
+   * @param {Registry} register - Registry instance for registering metrics.
+   */
+  constructor(
+    logger: Logger,
+    sdkClient: SDKClient,
+    mirrorNodeClient: MirrorNodeClient,
+    hbarLimiter: HbarLimit,
+    register: Registry,
+  ) {
+    this.logger = logger;
+    this.sdkClient = sdkClient;
+    this.hbarLimiter = hbarLimiter;
+    this.mirrorNodeClient = mirrorNodeClient;
+    this.consensusNodeClientHistogramCost = this.initCostMetric(register);
+    this.consensusNodeClientHistogramGasFee = this.initGasMetric(register);
+  }
+
+  /**
+   * Retrieves the transaction metrics for a given transaction ID
+   * by redirecting calls to either consensus node client or mirror node client based on default configuration.
+   *
+   * @param {string} transactionId - The ID of the transaction.
+   * @param {string} callerName - The name of the entity calling the transaction.
+   * @param {string} requestId - The request ID for logging purposes.
+   * @param {string} txConstructorName - The name of the transaction constructor.
+   * @param {string} operatorAccountId - The account ID of the operator.
+   */
+  public async captureTransactionMetrics(
+    transactionId: string,
+    callerName: string,
+    requestId: string,
+    txConstructorName: string,
+    operatorAccountId: string,
+    transactionType: string,
+    interactingEntity: string,
+  ) {
+    let gasUsed: number = 0;
+    let transactionFee: number = 0;
+    let txRecordChargeAmount: number = 0;
+    const formattedRequestId = formatRequestIdMessage(requestId);
+
+    // check if calls are default to consensus node or not
+    const defaultToConsensusNode = process.env.GET_RECORD_DEFAULT_TO_CONSENSUS_NODE === 'true';
+
+    // retrieve metrics
+    if (defaultToConsensusNode) {
+      const transactionRecordMetrics = await this.sdkClient.getTransactionRecordMetrics(
+        transactionId,
+        callerName,
+        requestId,
+        txConstructorName,
+        operatorAccountId,
+      );
+
+      if (transactionRecordMetrics) {
+        gasUsed = transactionRecordMetrics.gasUsed;
+        transactionFee = transactionRecordMetrics.transactionFee;
+        txRecordChargeAmount = transactionRecordMetrics.txRecordChargeAmount;
+      }
+    } else {
+      const transactionRecordMetrics = await this.mirrorNodeClient.getTransactionRecordMetrics(
+        transactionId,
+        callerName,
+        requestId,
+        txConstructorName,
+        operatorAccountId,
+      );
+
+      if (transactionRecordMetrics) {
+        transactionFee = transactionRecordMetrics.transactionFee;
+      }
+    }
+
+    // capture metrics to HBAR rate limiter and metric registry
+    if (transactionFee !== 0) {
+      this.addExpenseAndCaptureMetrics(
+        `TransactionExecution`,
```

Comment:
Why are those in these ` quotes but not normal quotes -> '

Reply:
oh it's jsut a habit they have the same function.